File: 09 - Py-Js/ques5.py
def str_freq(string):
    freq = {}
    for ch in string:
        if ch in freq:
            freq[ch] += 1
        else:
            freq[ch] = 1
    return freq

def replace_ch(string, Old, New):
    string = string.replace(Old, New)
    return string

def remove_first_ch(string, ch):
    index = string.find(ch)
    removed = string[:index] + string[index+1:]
    return removed
print(remove_first_ch("dev codes", "e"))

def remove_all_ch(string, ch):
    removed = string.replace(ch, "")
    return removed
print(remove_all_ch("dev codes", "e"))

def swap_ch(s4, s5, n):
    if n <= len(s4) and n <= len(s5):
        # swapping first n characters
        swapped_s4 = s5[:n] + s4[n:]
        swapped_s5 = s4[:n] + s5[n:]

        print("After swapping:")
        print("String 1:", swapped_s4)
        print("String 2:", swapped_s5)
    else:
        print("n is larger than the length of one of the strings!")

File: 09 - Py-Js/ques5.js
function str_freq(string){
    freq = {}
    for (let ch of string) {
        if (ch in freq) {freq[ch] += 1}
        else {freq[ch] = 1}
    }
    return freq
}

function replace_ch(string, Old, New){
    string = string.replace(Old, New)
    return string
}

function remove_first_ch(string, ch){
    let index = string.indexOf(ch)
    let removed = string.slice(0, index) + string.slice(index+ 1, string.length)
    return removed
}

function remove_all_ch(string, ch){
    removed = string.replaceAll(ch, "")
    return removed
}

function swap_ch(s1, s2, n){
    if (n <= s1.length && n <= s2.length){
        swapped_s1 = s2.slice(0,n) + s1.slice(n)
        swapped_s2 = s1.slice(0, n)+ s2.slice(n)
    }
    else return "n is larger than the length of one of the strings!"
    return [swapped_s1,swapped_s2]
}

console.log(swap_ch("mamta", "nikku", 5))
